qt/main.cpp

```diff
@@ -215,20 +214,19 @@ int main(int argc, char * argv[])
     fmt.setSwapInterval(1);
     fmt.setDepthBufferSize(16);
 
-    // This is a workaround for some systems,
-    // including MacOs and older Linux distros, which rely on X.org and Mesa.
-    // Where somehow the driver itself doesn't
-    // make all the otherwise supported GLSL versions available by default
-    // and such requests are somehow disregarded at later stages of execution.
-    // This setting here will be potentially overwritten and overruled anyway,
-    // and only needed to ensure that we have the needed GLSL compiler available
-    // later when we need it.
-    if (app.platformName() == QString("xcb") ||
-        app.platformName() == QString("cocoa"))
+    if (app.platformName() == QString("wayland"))
+    {
+      // TODO: Check if there is a way to use OpenGL 3.2 on wayland platform.
```

Comment:
This TODO comment is quite a bit misleading, because it suggests that "OpenGL 3.2 on wayland platform" does not work. Which is not true, I am using it daily, and nobody has complained about that in quite a while.

Reply:
Wait. If 3.2 is supported, why it doesn't work in OM then?

Reply:
It works with OM, I am using that every day.
If the hardware can do 3.2, then the appropriate code path is chosen in OM, if the hardware can only do ES 2.0, then that code path is chosen.

Reply:
I'm lost again.
1. Do you have several wayland devices?
2. One of them supports only OpenGL ES 2.0, right?
3. Another one works fine with OpenGL 3.2, right?
4. The code should not do an if for a platform name, but it should actually check which OpenGL/ES version is available, right? 
5. If (4) is `yes`, then what is the best way to do it?

Reply:
1. Yes
2. Yes
3. Yes
4. That's what it does currently [here](https://github.com/organicmaps/organicmaps/blob/master/qt/qt_common/map_widget.cpp#L352)
5. In theory what we have been doing.